From: SeongJae Park <sj@kernel.org>
Date: Thu, 8 Jan 2026 18:00:06 -0800
Subject: [PATCH] selftess/damon/wss_estimation: use access_memory_even instead
 of access_memory

Depending on running system's speed, the number of collected working set
size may vary.  Ensure it collects exactly 40 snapshots.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 ...fs_update_schemes_tried_regions_wss_estimation.py | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py b/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py
index a000982c59be..656e90cae74e 100755
--- a/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py
+++ b/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py
@@ -7,9 +7,9 @@ import time
 import _damon_sysfs
 
 def main():
-    # access two 10 MiB memory regions, 2 second per each
-    sz_region = 10 * 1024 * 1024
-    proc = subprocess.Popen(['./access_memory', '2', '%d' % sz_region, '2000'])
+    # repeately access two 5 MiB memory regions
+    sz_region = 5 * 1024 * 1024
+    proc = subprocess.Popen(['./access_memory_even', '4', '%d' % sz_region])
     kdamonds = _damon_sysfs.Kdamonds([_damon_sysfs.Kdamond(
             contexts=[_damon_sysfs.DamonCtx(
                 ops='vaddr',
@@ -36,11 +36,15 @@ def main():
 
         wss_collected.append(
                 kdamonds.kdamonds[0].contexts[0].schemes[0].tried_bytes)
+        if len(wss_collected) >= 40:
+            break
+    proc.terminate()
 
     wss_collected.sort()
     percentile = 75
     sample = wss_collected[int(len(wss_collected) * percentile / 100)]
-    error_rate = abs(sample - sz_region) / sz_region
+    expected_wss = sz_region * 2
+    error_rate = abs(sample - expected_wss) / expected_wss
     print('%d-th percentile (%d) error %f' %
             (percentile, sample, error_rate))
     acceptable_error_rate = 0.2
-- 
2.47.3

